src/components/Tables/RequestTable/RequestTable.tsx

```diff
@@ -136,21 +135,32 @@ const RequestTable: React.FC<RequestTableProps> = ({
 	/* Calls the middleware for approving a request
 		 This is passed to the Request modal */
 	const onApprove = async (request: DomainRequestAndContents) => {
+		setErrorMessage(undefined);
+		setShowLoadingIndicator(true); //displays loading indicator on overlay
 		try {
 			await staking.approveRequest(request);
 			setViewing(undefined);
 		} catch (e) {
 			// Catch thrown when user rejects transaction
 			console.error(e);
+			if (
+				e.message ===
```

Comment:
Something you could do here is check `e.code === 4001` instead of `e.message === "....."`

Reply:
Wasnt throwing any code when tested, only message, will give another try to catch it

Reply:
Changed tryFunction helper to include error code in message, and catch to search for "code: 4001". Since staking controller functions use that helper I was catching that and not the metamask one